build-system/tasks/sweep-experiments/index.js

```diff
@@ -0,0 +1,435 @@
+ * @return {!Array<string>}
+ */
+const filesContainingPattern = (dirs, string) =>
+  getStdoutLines(`grep -Elr "${cmdEscape(string)}" {${dirs.join(',')}}`);
+
+/**
+ * @param {string} fromHash
+ * @return {!Array<string>}
+ */
+const getModifiedSourceFiles = (fromHash) =>
+  getStdoutLines(`git diff --name-only ${fromHash}..HEAD | grep .js`).filter(
+    (file) => !globalWritablePaths.includes(file)
+  );
+
+/**
+ * Runs a jscodeshift transform under this directory.
+ * @param {string} transform
+ * @param {Array<string>=} args
+ * @return {string}
+ */
+const jscodeshift = (transform, args = []) =>
+  getStdout(
+    'npx jscodeshift ' +
+      ` -t ${__dirname}/jscodeshift/${transform} ` +
+      args.join(' ')
+  );
+
+/**
+ * @param {string} id
+ * @return {Array<string>} modified files
+ */
+function removeFromExperimentsConfig(id) {
+  jscodeshift('remove-experiment-config.js', [
+    `--experimentId=${id}`,
+    experimentsConfigPath,
+  ]);
+  return [experimentsConfigPath];
+}
+
+/**
+ * @param {!Object<string, *>} config
+ * @param {string} path
+ * @param {string} id
+ * @return {Array<string>} modified files
+ */
+function removeFromJsonConfig(config, path, id) {
+  delete config[id];
+  writeFileSync(path, JSON.stringify(config, null, 2) + '\n');
+  return [path];
+}
+
+/**
+ * @param {string} id
+ * @param {number} percentage
+ * @return {Array<string>} modified files
+ */
+function removeFromRuntimeSource(id, percentage) {
+  const possiblyModifiedSourceFiles = filesContainingPattern(
+    containRuntimeSource,
+    id
+  ).filter((name) => name.endsWith('.js'));
+  if (possiblyModifiedSourceFiles.length > 0) {
+    jscodeshift('remove-experiment-runtime.js', [
+      `--isExperimentOnLaunched=${percentage}`,
+      `--isExperimentOnExperiment=${id}`,
+      ...possiblyModifiedSourceFiles,
+    ]);
+  }
+  return possiblyModifiedSourceFiles;
+}
+
+/**
+ * @param {string} id
+ * @param {*} workItem
+ * @param {!Array<string>} modified
+ * @return {? string}
+ */
+function gitCommitSingleExperiment(id, workItem, modified) {
+  const message = workItem.previousHistory
+    .reduce(
+      (paragraphs, {hash, authorDate, subject}) => [
+        ...paragraphs,
+        `- ${hash} - ${authorDate} - ${subject}`,
+      ],
+      [
+        readableRemovalId(id, workItem),
+        `Previous history on ${prodConfigPath.split('/').pop()}:`,
+      ]
+    )
+    .join('\n\n');
+  getStdout(`git add ${modified.join(' ')}`);
+  return getStdout(`git commit -m "${cmdEscape(message)}"`);
+}
+
+function readableRemovalId(id, {percentage, previousHistory}) {
+  const {authorDate, hash} = previousHistory[0];
+  return `(${truncateYyyyMmDd(authorDate)}, ${hash}) \`${id}\`: ${percentage}`;
+}
+
+/**
+ * @param {!Array<stri  ng>=} files
+ */
+function maybeLintFix(files) {
+  if (argv.skip_lint_fix) {
+    log('❗️ (Not fixing lint issues due to --skip_lint_fix)');
+    return;
+  }
+  getOutput(`gulp lint --fix --files="${files.join(',')}"`);
```

Comment:
This is rather slow and I'm not sure why. Maybe we can plug into its internal utilities?